TFT/src/User/API/UI/GUI.c

```diff
@@ -606,49 +606,89 @@ void GUI_DispOne(int16_t sx, int16_t sy, const CHAR_INFO *pInfo)
 {
   if (pInfo->bytes == 0) return;
 
+  uint8_t w = pInfo->pixelWidth;
+  uint8_t h = pInfo->pixelHeight;
+  uint16_t bitMapSize = (h * w / 8);
+  uint8_t font[bitMapSize];
```

Comment:
It is not an optimization when you duplicate a variable knowing that it can be read from an already existing variable. Now for each character, the width and height are copied to a new variable instead of directly accessing the variable resulting in unnecessary CPU cycles.

Reply:
Yes, but they are used as conditions in loops (e.g. y < **h**;) or repeated more than one time in the code (e.g. sx + **w**, y * **w**). For that reason they have been loaded in new variables (e.g. avoiding access with **->** in loops etc...). Otherwise, we should be sure the compiler is able to optimize the code (e.g. loading and maintaining pInfo->pixelWidth in a register before it is used in the whole function).